bindings/java/src/main/native/Makefile.am

```diff
@@ -0,0 +1,46 @@
+JDK = $(JAVA_HOME)
+topdir=../../../../..
+javadir=$(topdir)/bindings/java
+MVN=@cd $(javadir) && mvn
+ucm_lib=ucm/.libs
+ucs_lib=ucs/.libs
+uct_lib=uct/.libs
+ucp_lib=ucp/.libs
+sources=bridge.cc request_util.cc
+
+# additional include paths necessary to compile the C library
+AM_CPPFLAGS = -fPIC -DPIC -Werror \
+		-I$(JDK)/include -I$(JDK)/include/linux \
+		-I$(topdir)/src/
+
+AM_CXXFLAGS = -std=c++11
+
+
+lib_LTLIBRARIES = libjucx.la
+
+libjucx_la_SOURCES = $(sources)
+
+libjucx_la_LDFLAGS = -shared -rdynamic -Wl,-rpath,${prefix}/lib
+
+libjucx_la_LIBADD = -L$(topdir)/src/$(ucm_lib) -L$(topdir)/src/$(ucs_lib) -L$(topdir)/src/$(uct_lib) -L$(topdir)/src/$(ucp_lib) -lucm -lucs -luct -lucp
+
+libjucx_la_DEPENDENCIES = Makefile.am Makefile.in Makefile
+
+
+#Compile Java source code and pack to jar
+all:
```

Comment:
$MVN will get executed regardless if ./configure detected or not java/mvn support.
the makefile should disqualify itself if no with-java selected.

Reply:
https://github.com/netanelyo/ucx/blob/0b20d076ef566d6f213c3e87022257d0c38329ed/Makefile.am#L44-L46
This should do, isn't it?

Reply:
yes, but it is not enough, you also need treat HAVE_MVN

Reply:
see http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/2919/label=ARM_test_node,worker=2/console

```
...
11:27:06 /bin/sh: mvn: command not found
11:27:06 make[2]: *** [clean] Error 127
11:27:06 make[2]: Leaving directory `/var/lib/jenkins/ucx.mellanox/gh-ucx-pr/label/ARM_test_node/worker/2/build-test/ucx-1.3.0/_build/bindings/java/src/main/native'
11:27:06 make[1]: *** [distclean-recursive] Error 1
11:27:06 make[1]: Leaving directory `/var/lib/jenkins/ucx.mellanox/gh-ucx-pr/label/ARM_test_node/worker/2/build-test/ucx-1.3.0/_build'
11:27:06 make: *** [distcheck] Error 1
```

Reply:
https://github.com/openucx/ucx/blob/5a490d456069d15d7b9f3aa0bd8aaa8f32527e95/config/m4/java.m4#L38-L40

Treating MVN here. If there is no mvn installed and 'with-java' option was selected, then an error will be thrown.